buildSrc/src/main/kotlin/WslJarPlugin.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ *  Copyright 2025 Budapest University of Technology and Economics
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.artifacts.ProjectDependency
+import org.gradle.api.plugins.JavaApplication
+import org.gradle.api.tasks.SourceSetContainer
+import org.gradle.kotlin.dsl.withType
+import java.io.File
+import java.util.jar.JarFile
+import java.util.jar.JarOutputStream
+
+/**
+ * Creates a WSL-compatible jar file that can be executed in Windows Subsystem for Linux (WSL).
+ * The plugin sets the class path in the jar manifest to use WSL-style relative paths.
+ *
+ * Use-case: run gradle build on Windows, then run the resulting WSL jar in WSL without rebuilding.
+ */
+class WslJarPlugin : Plugin<Project> {
+  override fun apply(project: Project) {
+    if (!System.getProperty("os.name").lowercase().contains("windows")) return
+
+    if (!project.plugins.hasPlugin("application")) {
+      throw IllegalStateException(
+        "WslJarPlugin requires the 'application' plugin to be applied."
+      )
+    }
+
+    val wslJar = project.tasks.register("wslJar") {
```

Comment:
It would probably be better to create a custom task type and use that here, instead of using a `doLast` lambda